pkg/config/config.go

```diff
@@ -36,6 +37,38 @@ func LoadConf(bytes []byte) (*sriovtypes.NetConf, error) {
 		return nil, fmt.Errorf("LoadConf(): VF pci addr is required")
 	}
 
+	// Get the vf pci address
+	vfPciAddress, err := utils.GetPciAddress(n.Master, n.VFID)
+	if err != nil {
+		return n, fmt.Errorf("error getting vf pci address for PF %s and VF id %d: %v", n.Master, n.VFID, err)
+	}
+
+	n.VFPciAddress = vfPciAddress
+
+	// Check if the device is already allocated.
+	// This is to prevent issues where kubelet request to delete a pod and in the same time a new pod using the same
+	// vf is started. we can have an issue where the cmdDel of the old pod is called AFTER the cmdAdd of the new one
+	// This will block the new pod creation until the cmdDel is done.
+	isAllocated, allocatedNs, err := utils.IsAllocated(vfPciAddress, DefaultCNIDir)
```

Comment:
we could roll the logic below into `IsAllocated` and simplify this method.

a VF is allocated if file exists and the namespace stored within it exists else it is not allocated.
if it is not allocated then IsAllocated can delete the stale file (as best effort operation) WDYT ?